### ISSUE TITLE
Fee increment adds Input and Change address without advertising

### ISSUE THREAD

**Issue by donob4n** [spesmilo/electrum#7322]
Hi,

When raising fee of a 1input x 1output transaction is relative easy to end with a 2input x 2output transaction without noticing which is terrible. First because if it is a 1x1 transaction it is very likely that the user wants to avoid change address and second because the fee raise will be also applied to a double sized transaction (I went from ~2$ to ~8$ when just wanted to go from 40sat to 60sat).

Considering that a fee raise scenario is pretty likely to be under some stress it would be nice to have a clear warn about you are doing and the final result.

**Comment by SomberNight** [comment]
Note the there are advanced options for bumping the fee. In your case the option you seem to want is "decrease payment".

Also note that that is not the default specifically because in a significant chunk of cases that is not what the user wants. E.g. if the user is paying a merchant, decreasing the payment output is highly likely going to invalidate the payment.

With the above in mind, what do you specifically suggest; or can the issue be closed?

**Comment by donob4n** [comment]
I think that some warn dialog like ("Your transaction will have more inputs!") should be nice. I got some testnet coins for try to improve this but the max button is not working properly.

Let me dig a little on it.

**Comment by donob4n** [comment]
How do you see something like this?


![raisefee](https://user-images.githubusercontent.com/35405566/120898274-7774c700-c62a-11eb-948f-f08dd61cda3b.png)

**Comment by SomberNight** [comment]
I don't like it. It is fairly common that we end up adding new inputs, and it is not dangerous or anything. Sure, some people might not want this, but keep in mind that many users probably don't even know what tx inputs/outputs are. For those people this warning is just scary, and without benefits.

Instead, try changing the text to: `(X new inputs will be added)`
Only display it if `X > 0`, and change the font colour from red to the default.
No need to mention the "advanced settings" option that is directly 20 pixels below.
Basically make it a short informational note, instead of scary warning.

**Comment by donob4n** [comment]
Ok, is not dangerous but you will end with pretty higher fee than expected. If you want to raise a 2$ fee transaction you expect something like 2,5$, 3$.... 4$... . A sat/byte increment of 50% is not expected to end in a 400% growth.

But I'm agree, just information message is nice and cleaner.

**Comment by donob4n** [comment]
Ey @SomberNight , could you take a look here?

**Comment by ecdsa** [comment]
Since the refactoring of the transaction creation flow (#8197), it now possible to view tx inputs and outputs interactively, when choosing/bumping the fee. This fixes this issue.
